### PR TITLE
Azurekid - Validate-AsimCsv function (ASIM)

### PR DESCRIPTION
**Pull request by azurekid**
Required items, please complete
   
   Change(s):
   - `Validate-ASimCsv.ps1`
   - `README.md`

   Reason for Change(s):
   - This script can be used to validate if no duplicate columns are added to the ASIM schema file.
     I have created this script because I noticed it occurs from time-to-time that duplicate columns 
     exists.

   Version Updated:
   - N/A
   
   Testing Completed:
   - [x] tested and validated the script including error handling.

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @azurekid,
Thank you for raising Pull Request with us! We will review it and provide you with an update. Thanks

**Comment by vakohl** [comment]
@azurekid This is great. I'll try to test it. I'll try to test it out this week.

File: ASIM/dev/ASimTester/Validate-ASimCsv/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# ASim Schema Validator
+
+This script validates if no duplicate columns are added the the ASim Schema.
```
Comment (vakohl) [inline ASIM/dev/ASimTester/Validate-ASimCsv/README.md:3]: please correct: the the

File: ASIM/dev/ASimTester/Validate-ASimCsv/Validate-ASimCsv.ps1 [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    foreach ($item in $schemaArray) {
+        $count = ($schemaArray | Where-Object ColumnName -eq $item.ColumnName).count
+        if ($count -gt 1) {
+            Write-Debug "Found duplicate column '$($item.ColumnName)' in schema '$($schema)'"
+            $duplicateColumns += $item
+        }
+    }
+}
+
+return $duplicateColumns | Sort-Object Schema, ColumnName, Class -Unique
```
Comment (vakohl) [inline ASIM/dev/ASimTester/Validate-ASimCsv/Validate-ASimCsv.ps1:50]: When duplicate fields exists, can we add a message at the top that describes the issue? Duplicate entries found for the following fields.
Also, when no duplicate entries, it should say no duplicate entries exists. csv is good. and exit

**Review by vakohl — CHANGES_REQUESTED** [review]

**Comment by v-atulyadav** [comment]
Hi @azurekid,
Hi @jayeshprajapaticrest, please check comments above from @vakohl and act accordingly. Thanks

**Comment by azurekid** [comment]
> Hi @azurekid, Hi @jayeshprajapaticrest, please check comments above from @vakohl and act accordingly. Thanks

Hi,

Will pick this up on 2023/11/08

File: ASIM/dev/ASimTester/Validate-ASimCsv/Validate-ASimCsv.ps1 (hunk 2 above)
Comment (azurekid) [reply ASIM/dev/ASimTester/Validate-ASimCsv/Validate-ASimCsv.ps1:50]: ### When the file contains no duplicates
image

### When the file contains duplicates
image

**Comment by azurekid** [comment]
Hi,

I have added a validation to see if any duplicate fields were detected and show a message in both scenario's 

Good suggestion btw 💯

**Comment by azurekid** [comment]
Hi All, 
Is there anything you are still waiting for from my side?
PR is now 3 months old.  Please let me know 🙏🏼

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**